src/components/document/document-content.tsx

```diff
@@ -53,25 +54,41 @@ export class DocumentContentComponent extends BaseComponent<IProps, IState> {
   private domElement: HTMLElement | null;
   private rowRefs: Array<TileRowComponent | null>;
   private mutationObserver: MutationObserver;
+  private scrollDisposer: IReactionDisposer;
 
   public componentDidMount() {
     if (this.domElement) {
       this.domElement.addEventListener("scroll", throttle(this.updateVisibleRows, 100));
       this.updateVisibleRows();
       this.scrollToSection(this.props.selectedSectionId);
 
-      if ((window as any).MutationObserver) {
-        this.mutationObserver = new MutationObserver(this.handleRowElementsChanged);
-        this.mutationObserver.observe(this.domElement, { childList: true });
-      }
       // We pass the domElement to our children, but it's undefined during the first render,
       // so we force an update to make sure we draw at least once after we have our domElement.
       this.forceUpdate();
+
+      this.scrollDisposer = reaction(() => this.stores.ui.scrollToTileId,
+        (tileId: string | undefined, prevTileId: string | undefined) => {
+        if (tileId) {
+          this.rowRefs.forEach((row: TileRowComponent | null) => {
+            if (row?.tileRowDiv && row.hasTile(tileId)) {
+              // Javascript struggles to scroll multiple elements at the same time,
+              // so we delay scrolling any document on the left and only animate the left document
```

Comment:
I'd note that the document icons/thumbnails are also 'readOnly' documents. So there could be a conflict if the browser is trying to scroll the full size doc and the thumbnail doc at the same time. I think there isn't a case where the full size readonly doc and the thumbnail doc are visible at the same time, so probably this isn't a problem in practice.